### ISSUE TITLE
[Elastic Agent] Check whether agent is installed before pursuing installation with "install -f" option

### ISSUE THREAD

**Issue by nimarezainia** [elastic/beats#29382]
**Describe the enhancement:**

Today when the elastic agent is being installed there's a check to see if one is already present and if so the installation is aborted. This is ideal as we don't want to install another agent and end up enrolling two  into Fleet (rendering one a zombie).

This check however doesn't exist when the "-f" (force) flag is used. The -f flag is there so that there's no verbosity which would cause any user scripts to fail. 

The request here to have the same behavior when the -f is used to install. In other words, with -f we check to see if the agent is already installed and abort the installation.

**Describe a specific use case for the enhancement or feature:**
We have had users using "install -f" multiple times which causes zombie agents created on the fleet. It's also a nicer work flow for users who otherwise would have to provide the logic outside of this install call to determine presence of the agent.

[@jlind23 ]

**Comment by ph** [comment]
If `-f` abort when the agent is already installed what is the purpose of **forcing** an installation? Some parts of the agent can `force` installed but not all of it?

**Comment by blakerouse** [comment]
I think the real issue is that `-f` is mixing the difference between force installation and non-interactive. I think what @nimarezainia is really wanting here is a new flag to basically say do not ask any questions, but abort if there is already an existing agent.

I suggest adding a `--non-interactive` and `-n` for this behavior and then changing the Fleet UI to use `-n`. I also like the `-n` because it reminds be of "No".

**Comment by ph** [comment]
Thanks @blakerouse,  I believe you are correct on this, `--non-interactive` really describes what is intended.
So we will probably need to do something like this.

- Add a non-interactive flag.
- Make `-f` behavior as `non-interactive` flag.
- Mark `-f` as deprecated?
- Update fleet UI to use `non-interactive`

**Comment by blakerouse** [comment]
@ph I don't believe `--non-interactive` replaces `-f`. I think they both should exist, each have a unique purpose.

`-f` is still useful, just not in the case that we want for Fleet UI. But for a local machine that I am developing on I use `-f` often.

**Comment by nimarezainia** [comment]
just to add to the comments here. Right now with -f we are pushing the logic to the admin to insert a "if agent is NOT installed, use   elastic-agent -f" else quit". What they do need is non-interactive or non-verbose. And we take care (as we do today) to check if it's installed or not.
